src/main/webapp/app/exercises/shared/exam-exercise-row-buttons/exam-exercise-row-buttons.component.html

```diff
@@ -96,15 +96,27 @@
             <span class="d-none d-md-inline" jhiTranslate="entity.action.re-evaluate">Re-evaluate</span>
         </a>
         <!-- Only show the the edit button if the exam has not ended. Disable the button if the exam is running -->
-        <a
-            *ngIf="!isExamOver()"
-            [class.disabled]="hasExamStarted()"
-            [routerLink]="['/course-management', course.id, 'exams', exam.id, 'exercise-groups', exerciseGroupId, exercise.type + '-exercises', exercise.id, 'edit']"
-            class="btn btn-warning btn-sm mr-1"
-        >
-            <fa-icon [icon]="'wrench'"></fa-icon>
-            <span class="d-none d-md-inline" jhiTranslate="entity.action.edit">Edit</span>
-        </a>
+        <div *ngIf="!isExamOver()" style="display: flex">
+            <!-- Display a warning next to the edit button if test run participations for this exercise exist -->
+            <fa-icon
+                style="margin: 0.3em"
+                [icon]="'exclamation-triangle'"
+                class="text-warning"
+                *ngIf="exercise.testRunParticipationsExist"
+                [ngbTooltip]="'artemisApp.quizExercise.edit.testRunSubmissionsExist' | artemisTranslate"
+                placement="auto"
+                container="body"
+            >
+            </fa-icon>
+            <a
+                [class.disabled]="hasExamStarted()"
```

Comment:
I think not using a function here would be better style. A variable which is initialized in the ngOnInit/ngOnChanges would probably be better.